gui/petitions.lua

```diff
@@ -0,0 +1,307 @@
+1                        = DemonicBinding
+2                        = Residency
+3                        = Citizenship
+4                        = Parley
+5                        = PositionCorruption
+6                        = PlotStealArtifact
+7                        = PromisePosition
+8                        = PlotAssassination
+9                        = PlotAbduct
+10                       = PlotSabotage
+11                       = PlotConviction
+12                       = Location
+13                       = PlotInfiltrationCoup
+14                       = PlotFrameTreason
+15                       = PlotInduceWar
+]]
+
+if not dfhack.world.isFortressMode() then return end
+
+-- from gui/unit-info-viewer.lua
+do -- for code folding
+--------------------------------------------------
+---------------------- Time ----------------------
+--------------------------------------------------
+local TU_PER_DAY = 1200
+--[[
+if advmode then TU_PER_DAY = 86400 ? or only for cur_year_tick?
+advmod_TU / 72 = ticks
+--]]
+local TU_PER_MONTH = TU_PER_DAY * 28
+local TU_PER_YEAR = TU_PER_MONTH * 12
+
+local MONTHS = {
+ 'Granite',
+ 'Slate',
+ 'Felsite',
+ 'Hematite',
+ 'Malachite',
+ 'Galena',
+ 'Limestone',
+ 'Sandstone',
+ 'Timber',
+ 'Moonstone',
+ 'Opal',
+ 'Obsidian',
+}
+Time = defclass(Time)
+function Time:init(args)
+ self.year = args.year or 0
+ self.ticks = args.ticks or 0
+end
+function Time:getDays() -- >>float<< Days as age (including years)
+ return self.year * 336 + (self.ticks / TU_PER_DAY)
+end
+function Time:getDayInMonth()
+ return math.floor ( (self.ticks % TU_PER_MONTH) / TU_PER_DAY ) + 1
+end
+function Time:getMonths() -- >>int<< Months as age (not including years)
+ return math.floor (self.ticks / TU_PER_MONTH)
+end
+function Time:getMonthStr() -- Month as date
+ return MONTHS[self:getMonths()+1] or 'error'
+end
+function Time:getDayStr() -- Day as date
+ local d = math.floor ( (self.ticks % TU_PER_MONTH) / TU_PER_DAY ) + 1
+ if d == 11 or d == 12 or d == 13 then
+  d = tostring(d)..'th'
+ elseif d % 10 == 1 then
+  d = tostring(d)..'st'
+ elseif d % 10 == 2 then
+  d = tostring(d)..'nd'
+ elseif d % 10 == 3 then
+  d = tostring(d)..'rd'
+ else
+  d = tostring(d)..'th'
+ end
+ return d
+end
+--function Time:__add()
+--end
+function Time:__sub(other)
+ if DEBUG then print(self.year,self.ticks) end
+ if DEBUG then print(other.year,other.ticks) end
+ if self.ticks < other.ticks then
+  return Time{ year = (self.year - other.year - 1) , ticks = (TU_PER_YEAR + self.ticks - other.ticks) }
+ else
+  return Time{ year = (self.year - other.year) , ticks = (self.ticks - other.ticks) }
+ end
+end
+--------------------------------------------------
+--------------------------------------------------
+end
+
+local we = df.global.ui.group_id
+
+local function getAgreementDetails(a)
+    local sb = {} -- StringBuilder
+
+    sb[#sb+1] = {{text = "Agreement #" ..a.id, pen = COLOR_RED}}
+    sb[#sb+1] = NEWLINE
```

Comment:
Recommendation:
```suggestion
    table.insert(sb, NEWLINE)
```
This would avoid a decent amount of repetition throughout this function.